lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -63,7 +63,15 @@ def handle_event_file(event_file)
 
       def handle_host_event(hostname, event)
         log_event("for host: #{hostname.inspect}", event)
-        publish_data_for(hostname, event['stdout'] + "\n", 'stdout') if event['stdout']
+        if event['stdout']
+          if @outputs[hostname]
+            publish_data_for(hostname, event['stdout'] + "\n", 'stdout')
+          elsif hostname == 'localhost'
+            broadcast_data(event['stdout'] + "\n", 'stdout')
+          else
+            raise "handle_host_event: unknown host #{hostname}"
```

Comment:
Since this PR was opened, we moved to processing the artifact files on demand. That means this raise does not really make sense anymore because the events may be processed after the job has already finished.

From where I stand we have essentially three options:
1) Broadcast events from unknown hosts the same way you do for localhost
2) Broadcast events from unknown hosts, but mark them as unexpected by broadcasting them as stderr
3) Just log those as errors